File: src/main/java/kitchenpos/menus/tobe/domain/MenuRepository.java
package kitchenpos.menus.tobe.domain;

import kitchenpos.products.tobe.domain.ProductId;

import java.util.List;
import java.util.Optional;

public interface MenuRepository {
    Menu save(Menu menu);

    Optional<Menu> findById(MenuId id);

    List<Menu> findAll();

    List<Menu> findAllByIdIn(List<MenuId> ids);

    List<Menu> findAllByProductId(ProductId productId);
}

File: src/main/java/kitchenpos/menus/tobe/domain/MenuRepository.kt
package kitchenpos.menus.tobe.domain

import java.util.*

fun MenuRepository.findByIdOrNull(menuId: UUID): Menu? = findById(menuId).orElse(null)

interface MenuRepository {
    fun save(menu: Menu): Menu

    fun findAllByProductId(productId: UUID): List<Menu>

    fun findById(menuId: UUID): Optional<Menu>

    fun findByIdIn(menuIds: List<UUID>): List<Menu>
}
